### ISSUE TITLE
A `try` body that returns unconditionally makes the whole `catch` and `finally` vanish

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#2]
**Type**: under-approximation, plus a distinct false positive on the same code.

## Repro

```js
try { return <div/>; } catch (e) { useEffect(() => { setN(n+1) }) }
```

reports `TryUncond (1 hooks)` instead of 2, plus `verified conditional-hook: all hooks run unconditionally`.
With `finally { setN(n+1) }` instead: `verified: no setter is called during render`.

## Cause

Both the handler and the finalizer are gated on `!builder.is_terminated()` and nothing starts a
fresh block afterwards, so the enclosing `lower_stmts` breaks out too. **In JS a `finally` always runs.**

The arm's own comment says the catch is walked "so hook extraction can find hooks inside catch
blocks", which is precisely what stops happening.

## Separate FP on the same code

When the gate *does* pass, catch and finally bodies are sequenced **unconditionally** after the try
body in one chain, so `ExitDominance` / `must_setter_on_all_paths` are told a catch-only write
happens on every path.

*Location: `lowering/cfg_builder.rs`, `TryStatement`.*

**Comment by rboudrouss** [comment]
Fixed in `3a068ed`. Both halves.

## Cause, confirmed exactly as written

The lowering ran the three bodies in sequence, each gated on `!builder.is_terminated()`. A `try` whose body returns terminates the block, so the whole `catch` and `finally` were never lowered — while the arm's own comment said it walked the catch "so hook extraction can find hooks inside catch blocks".

## Fix

A branch on an unknowable condition — the body may or may not throw — says both true things at once: the handler is on *a* path and not on all of them, and both arms converge on the finalizer, which is on all of them. That is what `try`/`catch`/`finally` means.

```
        ┌── Branch(⊤) ──┐
        ▼               ▼
    [try body]     [catch body]
        └──────┬────────┘
               ▼
        [finally body]
```

## Measured on both repros, plus the FP

| shape | before | after |
|---|---|---|
| `useEffect` in a catch after `return` | invisible, 1 hook, `✓` | `conditional-hook` (Error) + `infinite-loop` |
| `setN` in a `finally` after `return` | invisible, `✓` | `setter-in-render` |
| `setN` only in the catch | **Error** — claimed every-path | **Warning** |
| `setN` with no `try` (control) | Error | Error |

The third row is the separate FP this issue describes. The fourth is what proves the demotion comes from the `try` rather than from breaking the Error tier.

Worth noting the `conditional-hook` in row 1: a hook inside a `catch` genuinely violates the rules of hooks, and the old straight-line lowering had `conditional-hook` issuing the *opposite* assurance — "all hooks run unconditionally".

## Deliberate divergence

A `return` inside the try body seals its block, so that path does not reach the finalizer, where JS would run it first. The finalizer stays reachable through the throwing arm, so its hooks and writes are still found; what is lost is only its presence on the returning path, which costs `must`-strength (an Error demoted to a Warning) and never a finding. That is why `FinallyAfterReturn` reports a Warning. Recorded in `docs/limitations.md`.

## Corpus

1340 → 1344 (+4, **0 removed**) — the fix only opens code that was never lowered, it closes none.

All four additions are `missing-deps` on `t`, twenty's Lingui macro, imported at module level and therefore constant across renders. That family is **not created here**: 208 such rows before this change, 212 after. The reads simply sat in catch blocks nobody lowered. Filed as #141, along with the note that it did not reduce to a minimal repro.

Five regression tests in `tests/try_catch_finally.rs`, gated by removal: restoring the straight-line lowering reddens exactly four and leaves the `AlwaysWrite` control green.
